### PR TITLE
Enabled cors and container Generation via ARM

### PR DESCRIPTION
**Pull request by mwevill**
Added functionality to use new version of the platform building block to enable default cors handling.
Moved container creation from separate script to within ARM.
Reused Variables from current scripts to ensure no changes to environmental functionality

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -54,6 +54,16 @@
         },
         "originPath": {
             "type": "string"
+        },
+        "enableCors": {
+            "type": "string",
+            "defaultValue": "False"
+        },
+        "containerName": {
+            "type": "securestring"
```
Comment (jack-education) [inline azure/template.json]:
```suggestion
            "type": "string"
```

File: azure/template.json [hunk 2]
```diff
@@ -54,6 +54,16 @@
...
+        },
+        "ContainerPermission": {
```
Comment (jack-education) [inline azure/template.json]:
```suggestion
        "containerPermission": {
```

**Comment by mwevill** [comment]
> Could an array of containers be deployed instead, for greater flexbility when adding new containers?
> 
> Example: https://github.com/SkillsFundingAgency/das-pensionsregulator/blob/master/azure/template.json#L361-L392

I don't see it being necessary for this piepline. It's a CDN with 1 endpoint.

**Comment by mwevill** [comment]
And we are just replicating the functionality that is currently scripted.

File: azure/template.json [hunk 3]
```diff
@@ -77,11 +87,38 @@
...
+                },
+                "parameters": {
+                    "storageAccountName": {
+                        "value":  "[parameters('storageAccountName')]"
+                    },
+                    "containerName": {
+                        "value": "[parameters('containerName')]"
+                    },
+                    "publicAccess": {
+                        "value": "[parameters('ContainerPermission')]"
```
Comment (jack-education) [inline azure/template.json]:
```suggestion
                        "value": "[parameters('containerPermission')]"
```

File: azure/template.json [hunk 4]
```diff
@@ -77,10 +84,45 @@
...
+                "parameters": {
+                    "storageAccountName": {
+                        "value": "[parameters('storageAccountName')]"
+                    },
+                    "containerName": {
+                        "value": "[parameters('containerArray')[copyIndex()].name]"
+                    },
+                    "publicAccess": {
+                        "value": "[parameters('containerArray')[copyIndex()].publicAccess]"
+                    }
```
Comment (jack-education) [inline azure/template.json]: Could publicAccess be hardcoded to be None?

File: azure/template.json [hunk 5]
```diff
@@ -77,10 +84,45 @@
...
+        },
+        {
+            "apiVersion": "2017-05-10",
+            "name": "[concat('storage-container', parameters('containerArray')[copyIndex()].name)]",
+            "type": "Microsoft.Resources/deployments",
+            "condition": "[greater(length(parameters('containerArray')), 0)]",
+            "properties": {
+                "mode": "Incremental",
+                "templateLink": {
+                    "uri": "https://raw.githubusercontent.com/SkillsFundingAgency/das-platform-building-blocks/master/templates/storage-container.json",
```
Comment (jack-education) [inline azure/template.json]: Could a deploymentUrlBase variable be used?

File: azure/template.json (hunk 4 above)
Comment (mwevill) [reply azure/template.json]: No Because in this end it's Blob level as it's the source for the CDN

File: azure/template.json (hunk 5 above)
Comment (mwevill) [reply azure/template.json]: Possibly this is a a task to replace a complex poweshell script. Not a complete re-implementation of the service

**Pull request merged**